lib/dependencies.sh

```diff
@@ -326,3 +326,28 @@ npm_prune_devdependencies() {
     meta_set "skipped-prune" "false"
   fi
 }
+
+pnpm_prune_devdependencies() {
+  local pnpm_version
+  local build_dir=${1:-}
+
+  pnpm_version=$(pnpm --version)
+
+  if [ "$NODE_ENV" == "test" ]; then
+    echo "Skipping because NODE_ENV is 'test'"
+    meta_set "skipped-prune" "true"
+    return 0
+  elif [ "$NODE_ENV" != "production" ]; then
+    echo "Skipping because NODE_ENV is not 'production'"
+    meta_set "skipped-prune" "true"
+    return 0
+  elif [ -n "$NPM_CONFIG_PRODUCTION" ]; then
+    echo "Skipping because NPM_CONFIG_PRODUCTION is '$NPM_CONFIG_PRODUCTION'"
+    meta_set "skipped-prune" "true"
+    return 0
```

Comment:
It is this customization that might be worth documenting. Although as it follows what the Heroku official package for `npm` does it might be ok leaving as is.

Reply:
That's why I left it like that, so it would have maximum backwards compatibility. But perhaps I could change it to something like `PNPM_SKIP_PRUNE`. Otherwise, it behaves almost exactly the same as the NPM prune command, so I think it would be okay to leave it as is.